### PR TITLE
indirection: get new routeToken if serverId mismatch

### PR DESCRIPTION
**Pull request by Rash419**
Change-Id: I0efb38b714894dffb44cb5e2ca06be12ccf43b89

* Target version: master 

Attached a screencast:
[Screencast from 2023-08-28 19-35-34.webm](https://github.com/CollaboraOnline/online/assets/26241069/f506566d-bb37-4666-99f0-f2b438c10829)

After 3 retries it will open the document but with warning modal dialog

### CODE REVIEW

File: browser/src/core/Socket.js [hunk 1]
```diff
@@ -576,6 +577,35 @@ app.definitions.Socket = L.Class.extend({
 					setTimeout(reloadFunc, 5000);
 				}
 			}
+			if (window.indirectSocket) {
+				if (window.expectedServerId != this.WSDServer.Id) {
+					if (this.IndirectSocketReconnectCount++ >= 3) {
+						this._map.uiManager.showInfoModal('wrong-server-modal', _('Cluster configuration warning'), _('Collaborative editing or copy/paste might not work correctly, there are chances of potential document split'), '', _('OK'), null, false);
```
Comment (mmeeks) [inline browser/src/core/Socket.js:583]: "Your %product% cluster appear to be mis-configured or scaling rapidly - please contact your system administrator. Continuing with editing may result in multiple users not seeing each other, conflicts in the document storage and/or copy/paste problems."

**Review by mmeeks — COMMENTED** [review]
Looks plausible - but - we should tweak the message I think. Having more details on what is wrong may be helpful below: "expected server_id X for route_token Y but connected to Z" - we want to be able to debug this easily.

File: browser/src/core/Socket.js (hunk 1 above)
Comment (Rash419) [reply browser/src/core/Socket.js]: done, thanks

**Comment by Rash419** [comment]
Screenshot from 2023-08-28 20-33-06

**Pull request merged**